application_test.go

```diff
@@ -65,19 +65,76 @@ func TestApplicationCPU(t *testing.T) {
 
 func TestApplicationArgs(t *testing.T) {
 	app := NewDockerApplication()
-	assert.Equal(t, 0, len(app.Args))
-	app.Arg("-p").Arg("option", "-v")
-	assert.Equal(t, 3, len(app.Args))
-	assert.Equal(t, "-p", app.Args[0])
-	assert.Equal(t, "option", app.Args[1])
-	assert.Equal(t, "-v", app.Args[2])
+	assert.Nil(t, app.Args)
+	app.AddArgs("-p").AddArgs("option", "-v")
+	assert.Equal(t, 3, len(*app.Args))
+	assert.Equal(t, "-p", (*app.Args)[0])
+	assert.Equal(t, "option", (*app.Args)[1])
+	assert.Equal(t, "-v", (*app.Args)[2])
+
+	app.EmptyArgs()
+	assert.NotNil(t, app.Args)
+	assert.Equal(t, 0, len(*app.Args))
+}
+
+func ExampleApplication_AddConstraint() {
```

Comment:
It's an example that will show up as an example in godoc. The reason I added an example here is the kind of weird interface marathon provides for contraints: It's an array of values which can differ in length. The hope is that an example clarifies this.